Sources/MongoKitten/AggregateStage.swift

```diff
@@ -36,9 +39,26 @@ public struct AggregateBuilderStage {
     }
     
     public static func project(_ projection: Projection) -> AggregateBuilderStage {
+        func valueUnwrapper(_ value: Primitive) -> WireVersion?{
+            switch value {
+            case let value as Int32:
+                if value == 0 { return .mongo3_4 } // indicates excluded fields
+                return nil
+            case let value as String:
+                if value == "$$REMOVE" { return .mongo3_6 } // indicates conditionally excluded fields
+                return nil
+            case let value as Document:
+                return valueUnwrapper(value)
```

Comment:
I can guarantee you that this line of code will crash. This function will infinitely recurse into calling itself. You're looking to unwrap the values.